blogposts/2021/release-3.31.md

```diff
@@ -0,0 +1,88 @@
+---
+title: "Sourcegraph 3.31 release"
+publishDate: 2021-08-20T10:00-07:00
+description: "Sourcegraph 3.31 introduces UPDATE"
+tags: [blog, release]
+slug: "release/3.31"
+published: false
+heroImage: 
+socialImage: 
+changelogItems:
+  - description: "Backend Code Insights GraphQL queries now support arguments `includeRepoRegex` and `excludeRepoRegex` to filter on repository names."
+    url: https://github.com/sourcegraph/sourcegraph/pull/23256
+    category: Code Insights
+  - description: Operator documentation has been added to the Search Reference sidebar section.
+    url: https://github.com/sourcegraph/sourcegraph/pull/23116
+    category: Search
+  - description: Syntax highlighting support for the Cue language.
+    url: https://cuelang.org
+    category: Search
+  - description: Reintroduced a revised version of the Search Types sidebar section.
+    url: https://github.com/sourcegraph/sourcegraph/pull/23170
+    category: Search
+  - description: "Added support for [`git p4`'s `--use-client-spec` option](https://git-scm.com/docs/git-p4#Documentation/git-p4.txt---use-client-spec) can now be enabled by configuring the `p4.client` field in the Perforce integration."
+    url: https://github.com/sourcegraph/sourcegraph/pull/23845
+    category: Admin
+  - description: "Batch Changes changesets can now be [set to published when previewing new or updated batch changes](https://docs.sourcegraph.com/batch_changes/how-tos/publishing_changesets#within-the-ui)."
+    url: https://github.com/sourcegraph/sourcegraph/issues/22912
+    category: Batch Changes
+  - description: "Code Insights drill-down filters now allow filtering insights data on the dashboard page using repo: filters."
+    url: https://github.com/sourcegraph/sourcegraph/issues/23186
+    category: Code Insights
+  - description: "Code Insights feature flag `DISABLE_CODE_INSIGHTS` environment variable has moved from the `repo-updater` service to the `worker` service. Any users of this flag will need to update their `worker` service configuration to continue using it."
```

Comment:
@iskyOS @Joelkw - Thoughts on cutting this?

Reply:
I'd be OK with it; @Joelkw do you anticipate many users will need to make adjustments for this?

Reply:
I know that we have 2-4 customer instances this will directly affect – I think it may be worth putting it here so they're more likely to see it, if we're okay keeping it in. Regardless, I'll also update CE but they may choose to upgrade before working with a CE and hit this after the fact.